### PR TITLE
WIP trigger on leave script

### PR DESCRIPTION
**Pull request by pau-tomas**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](/chrismaltby/gb-studio/blob/develop/.github/COMMIT_MESSAGE_GUIDELINES.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature

* **What is the current behavior?** (You can also link to an open issue here)

Triggers have one script that execute when the player steps into them

* **What is the new behavior (if this is a feature change)?**

Added a new script for triggers that executes when the player steps out of them

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

Yes. A migration script needs to be added for old triggers to have an empty "on leave" script. Otherwise the app crashes when selecting the trigger. 

* **Other information**:

I submitted a PR with the changes for gbvm and a sample project https://github.com/chrismaltby/gbvm/pull/107

### CODE REVIEW

File: src/store/features/entities/entitiesTypes.ts [hunk 1]
```diff
@@ -99,7 +99,7 @@ export type Trigger = {
   width: number;
   height: number;
   script: string[];
-  leaveScript: string[];
+  leaveScript?: string[]; // Optional: Old versions of GBS don't have this script
```
Comment (pau-tomas) [inline src/store/features/entities/entitiesTypes.ts:102]: @chrismaltby let me know if you prefer a different way to deal with this case. Maybe a migration script?

File: appData/src/gb/include/gbs_types.h [hunk 2]
```diff
@@ -1,9 +1,7 @@
 #ifndef GBS_TYPES_H
 #define GBS_TYPES_H
 
-#ifdef CGB
-    #include <gb/cgb.h>
-#endif
+#include <gb/cgb.h>
```
Comment (chrismaltby) [inline appData/src/gb/include/gbs_types.h:4]: Is this ifdef removal needed for the feature?

File: src/store/features/entities/entitiesTypes.ts (hunk 1 above)
Comment (chrismaltby) [reply src/store/features/entities/entitiesTypes.ts:102]: Yeah I think a migration script so we can expect all triggers to at least have an empty array would be preferable. With the latest v3beta branch you no longer need an "EVENT_END" event in the array for it work so a migration to just add `leaveScript: []` to every trigger should be enough :-)

Comment (chrismaltby) [reply src/store/features/entities/entitiesTypes.ts:102]: v3alpha even, you know what I mean :-)

File: appData/src/gb/include/gbs_types.h (hunk 2 above)
Comment (pau-tomas) [reply appData/src/gb/include/gbs_types.h:4]: Ooops not sure why it's here

**Pull request merged**